xarray/backends/api.py

```diff
@@ -1230,6 +1231,7 @@ def to_netcdf(
     compute: bool = True,
     multifile: Literal[False] = False,
     invalid_netcdf: bool = False,
+    auto_complex: bool | None = None,
```

Comment:
What does `auto_complex = None` mean? Is it the same as `auto_complex = False`?
EDIT: Ah, to avoid passing the argument to `h5netcdf`? But `invalid_netcdf` is just a plain `bool`, so is that handled differently for `netCDF4`?

Should we also consider defaulting to `True`? I'm definitely interested in finding out if there are any cases where this results in unwanted behaviour

Reply:
Yeah, not sure, if I handled that correctly. It's more or less a security measure against feeding the kwarg to versions which can't handle those yet. I think this was the simplest solution without adding a whole bunch of boilerplate code. But maybe there is an easier solution to that.